kratos/includes/properties.h

```diff
@@ -91,6 +91,7 @@ class Properties : public IndexedObject
 
     typedef std::unordered_map<std::size_t, TableType> TablesContainerType; // This is a provisional implmentation and should be changed to hash. Pooyan.
 
+    typedef std::unordered_map<IndexType, Properties::Pointer> SubPropertiesListType;
```

Comment:
here i would use the Properties object, so to make it exactly identical to one level up

Reply:
Do you mean instead of the pointer?